helix-term/src/application.rs

```diff
@@ -116,24 +123,37 @@ impl Application {
             // Unset path to prevent accidentally saving to the original tutor file.
             doc_mut!(editor).set_path(None)?;
         } else if !args.files.is_empty() {
-            let first = &args.files[0]; // we know it's not empty
+            let first = &args.files[0].0; // we know it's not empty
             if first.is_dir() {
                 std::env::set_current_dir(&first)?;
                 editor.new_file(Action::VerticalSplit);
                 compositor.push(Box::new(ui::file_picker(".".into())));
             } else {
                 let nr_of_files = args.files.len();
                 editor.open(first.to_path_buf(), Action::VerticalSplit)?;
-                for file in args.files {
+                for (file, row, col) in args.files {
                     if file.is_dir() {
                         return Err(anyhow::anyhow!(
                             "expected a path to file, found a directory. (to open a directory pass it as first argument)"
                         ));
                     } else {
-                        editor.open(file.to_path_buf(), Action::Load)?;
+                        let doc_id = editor.open(file, Action::Load)?;
+                        // with Action::Load all documents have the same view
+                        let view_id = editor.tree.focus;
+                        let doc = editor.document_mut(doc_id).unwrap();
+                        let row = row.min(doc.text().len_lines() - 1);
+                        let pos = Selection::point(
+                            (doc.text().line_to_char(row) + col)
```

Comment:
`ropey` exposes a fallible version of `line_to_char` as `try_line_to_char` -- maybe you could use that and if it fails, unwrap to the default selection at `(0,0)`?

If you try to open a file to an obviously incorrect line (e.g. `hx README.md:100000000:1`), helix will panic because it's obviously out of bounds.

Reply:
I came up with a minor diff that would implement this functionality (but instead of treating an over-long index as 0, it just goes to the end of the file, which I think is more correct):

```diff
diff --git a/helix-term/src/application.rs b/helix-term/src/application.rs
index 2a86cff..c5ab2aa 100644
--- a/helix-term/src/application.rs
+++ b/helix-term/src/application.rs
@@ -1,4 +1,7 @@
-use helix_core::{line_ending::line_end_char_index, merge_toml_values, syntax, Selection};
+use helix_core::{
+    line_ending::{line_end_char_index, rope_end_without_line_ending},
+    merge_toml_values, syntax, Selection,
+};
 use helix_lsp::{lsp, util::lsp_pos_to_pos, LspProgressMap};
 use helix_view::{theme, Editor};
 
@@ -141,11 +144,14 @@ impl Application {
                         // with Action::Load all documents have the same view
                         let view_id = editor.tree.focus;
                         let doc = editor.document_mut(doc_id).unwrap();
-                        let pos = Selection::point(
-                            (doc.text().line_to_char(row) + col)
-                                .min(line_end_char_index(&doc.text().slice(..), row)),
-                        );
-                        doc.set_selection(view_id, pos);
+                        let selection = if let Ok(pos) = doc.text().try_line_to_char(row) {
+                            Selection::point(
+                                (pos + col).min(line_end_char_index(&doc.text().slice(..), row)),
+                            )
+                        } else {
+                            Selection::point(rope_end_without_line_ending(&doc.text().slice(..)))
+                        };
+                        doc.set_selection(view_id, selection);
                     }
                 }
                 editor.set_status(format!("Loaded {} files.", nr_of_files));

```

Reply:
> If you try to open a file to an obviously incorrect line (e.g. hx README.md:100000000:1), helix will panic because it's obviously out of bounds.

I tested this but it doesn't panic. I did fixed the panics previously already.

Reply:
Oh, indeed, looks like I was on an old commit. Sorry for the noise!